packages/scaffold/src/views/w3m-convert-select-token-view/index.ts

```diff
@@ -0,0 +1,261 @@
+    clearInterval(this.interval)
+  }
+
+  // -- Render -------------------------------------------- //
+  public override render() {
+    return html`
+      <wui-flex flexDirection="column" gap="s">
+        ${this.templateSearchInput()} ${this.templateSuggestedTokens()} ${this.templateTokens()}
+      </wui-flex>
+    `
+  }
+
+  // -- Private ------------------------------------------- //
+  private watchTokens() {
+    this.interval = setInterval(() => {
+      ConvertController.getNetworkTokenPrice()
+      ConvertController.getMyTokensWithBalance()
+    }, 5000)
+  }
+
+  private onSelectToken(token: TokenInfo) {
+    if (this.targetToken === 'sourceToken') {
+      ConvertController.setSourceToken(token)
+    } else {
+      ConvertController.setToToken(token)
+    }
+    RouterController.goBack()
+  }
+
+  private templateSearchInput() {
+    return html`
+      <wui-flex class="search-input-container" gap="xs">
+        <wui-input-text
+          class="network-search-input"
+          size="sm"
+          placeholder="Search token"
+          icon="search"
+          .value=${this.searchValue}
+          @inputChange=${this.onSearchInputChange.bind(this)}
+        ></wui-input-text>
+      </wui-flex>
+    `
+  }
+
+  private templateTokens() {
+    const yourTokens = ConvertController.state.myTokensWithBalance
+      ? Object.values(ConvertController.state.myTokensWithBalance)
+      : []
+    const tokens = ConvertController.state.popularTokens
+      ? Object.values(ConvertController.state.popularTokens)
+      : []
+
+    const filteredYourTokens: TokenInfoWithBalance[] = this.filterTokensWithText<
+      TokenInfoWithBalance[]
+    >(yourTokens, this.searchValue)
+    const filteredTokens = this.filterTokensWithText<TokenInfo[]>(tokens, this.searchValue)
+
+    return html`
+      <wui-flex class="tokens-container">
+        <wui-flex class="tokens" flexDirection="column">
+          ${filteredYourTokens?.length > 0
+            ? html`
+                <wui-flex justifyContent="flex-start" padding="s">
+                  <wui-text variant="paragraph-500" color="fg-200">Your tokens</wui-text>
+                </wui-flex>
+                ${filteredYourTokens.map(tokenInfo => {
+                  const selected =
+                    tokenInfo.symbol === this.sourceToken?.symbol ||
+                    tokenInfo.symbol === this.toToken?.symbol
+
+                  return html`
+                    <wui-token-list-item
+                      name=${tokenInfo.name}
+                      ?disabled=${selected}
+                      symbol=${tokenInfo.symbol}
+                      price=${tokenInfo?.price}
+                      amount=${tokenInfo?.balance}
+                      imageSrc=${tokenInfo.logoURI}
+                      @click=${() => {
+                        if (!selected) {
+                          this.onSelectToken(tokenInfo)
+                        }
+                      }}
```

Comment:
you could avoid inlining this by currying the parameter and adding selected there:

```
const onSelectToken = (tokenInfo: TokenInfo, selected: boolean) => () => {...}

@click={this.onSelectToken(tokenInfo, selected)}
```